release.sh

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env bash
+
+# Copyright © 2021 The Tekton Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+RELEASE_VERSION=""
+
+declare -r SCRIPT_PATH=$(readlink -f "$0")
+declare -r SCRIPT_DIR=$(cd $(dirname "$SCRIPT_PATH") && pwd)
+declare -r API_DIR="$SCRIPT_DIR/api"
+declare -r UI_DIR="$SCRIPT_DIR/ui"
+declare -r RELEASES_DIR="$SCRIPT_DIR/releases"
+declare -r UPSTREAM_REMOTE="origin"
+
+declare -a BINARIES=(
+  git
+  hub
+)
+
+info() {
+  echo "INFO: $@"
+}
+
+err() {
+  echo "ERROR: $@"
+}
+
+getReleaseVersion() {
+  [[ -z ${RELEASE_VERSION} ]] && {
+    read -r -e -p "Enter a target release (i.e: v0.1.2): " RELEASE_VERSION
+    [[ -z ${RELEASE_VERSION} ]] && {
+      echo "no target release"
+      exit 1
+    }
+  }
+  [[ ${RELEASE_VERSION} =~ v[0-9]+\.[0-9]*\.[0-9]+ ]] || {
+    echo "invalid version provided, need to match v\d+\.\d+\.\d+"
+    exit 1
+  }
+}
+
+buildDbMigrationImage() {
+  info Building DB Migration Image
+  echo -----------------------------------
+  cd "$API_DIR"
+  docker build -f db.Dockerfile -t quay.io/tekton-hub/db-migration:${RELEASE_VERSION} . && docker push quay.io/tekton-hub/db-migration:${RELEASE_VERSION}
+  info DB Migration Image Build Successfully
+  echo -----------------------------------
+}
+
+buildApiImage() {
+  info Building API Image
+  echo -----------------------------------
+  cd "$API_DIR"
+  docker build -t quay.io/tekton-hub/api:${RELEASE_VERSION} . && docker push quay.io/tekton-hub/api:${RELEASE_VERSION}
+  info API Image Build Successfully
+  echo -----------------------------------
+}
+
+buildUiImage() {
+  info Building UI Image
+  echo -----------------------------------
+  cd "$UI_DIR"
+  docker build -t quay.io/tekton-hub/ui:${RELEASE_VERSION} . && docker push quay.io/tekton-hub/ui:${RELEASE_VERSION}
```

Comment:
should we parameterise the docker command as there can be a case when one doesn't have docker installed but has podman on their machine? Maybe we can keep the default as docker and one change the param value on the fly if they want to do so

Reply:
Yup, makes sense